File: Backend/SOS-Clean-House-back/src/main/java/com/SOS_Clean_House_back/model/Horario.java
package com.SOS_Clean_House_back.model;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@AllArgsConstructor @NoArgsConstructor @Data @Builder
public class Horario {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Integer id;
    private String horarioInicio;
    private String horarioFim;
    private Boolean meioPeriodo;

    @JsonProperty(access = JsonProperty.Access.WRITE_ONLY)
    @OneToOne
    @JoinColumn(name = "servico_id")
    private Servico servico;

    @ManyToOne
    @JoinColumn(name = "prestador_id")
    private Prestador prestador;

}

File: Frontend/SOS-Clean-House-front/src/app/componentes/interface/Horario.ts
import { Prestador } from "./Prestador";
import { Servico } from "./Servico";

export interface Horario {
    id: number;
    horarioInicio: String;
    horarioFim: String;
    meioPeriodo: boolean;
    servico: Servico;
    prestador: Prestador;
  }